core/src/main/java/com/datastrato/gravitino/lock/LockManager.java

```diff
@@ -0,0 +1,119 @@
+/*
+ * Copyright 2023 Datastrato Pvt Ltd.
+ * This software is licensed under the Apache License version 2.
+ */
+
+package com.datastrato.gravitino.lock;
+
+import com.datastrato.gravitino.NameIdentifier;
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.collect.Maps;
+import java.util.Map;
+import java.util.Stack;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class LockManager {
+  public static final Logger LOG = LoggerFactory.getLogger(LockManager.class);
+
+  private static final NameIdentifier ROOT = NameIdentifier.ROOT;
+  private final Map<NameIdentifier, LockNode> lockNodeMap = Maps.newConcurrentMap();
```

Comment:
How to release the unused locks.

Reply:
We can use the Guava cache to refresh the map and release locks that have been visited for a long time.